sonic-net/sonic-buildimage
# [action] [PR:15239] [Nokia][sonic-platform] Update Nokia sonic-platform submodule

This fixes https://github.com/Nokia-ION/ndk/issues/16.

#### Why I did it
1. To support dynamic swapping of module types/speeds (400G/100G/40G)
2. To optimize CMIS ZR optics operation

#### How I did it

Reinitialize xcvr_api at module removal/insertion time, and also optimize cache for ZR optics.

#### How to verify it

1. Verify that different (supported) module types can be dynamically swapped (removed/inserted) and that each is properly provisioned by Xcvrd and has its EEPROM information accurately reported in Redis DB (using "show transceiver eeprom") as well as "sfputil show eeprom" direct access. 

2. Also verify that Xcvrd initialization and operation with 400G CMIS ZR optics is both efficient and functional.
 LICENSE MAINTAINERS Makefile Makefile.cache Makefile.work README.buildsystem.md README.md ThirdPartyLicenses.txt azure-pipelines.yml build_debian.sh build_debug_docker_j2.sh build_docker.sh build_image.sh check_install.py device dockers files functions.sh get_docker-base.sh install_sonic.py installer onie-image-arm64.conf onie-image-armhf.conf onie-image.conf onie-mk-demo.sh platform push_docker.sh rules scripts slave.mk sonic-slave-bullseye sonic-slave-buster sonic-slave-jessie sonic-slave-stretch src update_screen.sh edit 6/14/23: pushed enhanced caching (full memory map) support and elimination of base class APIs override.

#### Which release branch to backport (provide reason below if selected)


#### Description for the changelog

Cache tuning for ZR optics and add support for dynamic module type/speed swapping.
